### PR TITLE
fix(js_formater): strip useless quotes

### PR DESCRIPTION
**Pull request by suxin2017**
## Summary

fix(js_formater): strip useless quotes
#3484

```diff
- import J from "file.json" with { "type": "json" }
+ import J from "file.json" with { type: "json" }
```

### CODE REVIEW

**Comment by Conaclos** [comment]
Thanks! Could you rebase on main? I've just updated the Prettier snapshots.

**Comment by suxin2017** [comment]
I have modified it, but I did not modify the code related to paser, I wonder why the performance has decreased

**Comment by ematipico** [comment]
> I have modified it, but I did not modify the code related to paser, I wonder why the performance has decreased

It's a flaky test, it's unrelated to your changes

File: crates/biome_js_formatter/src/utils/string_utils.rs [hunk 1]
```diff
@@ -14,6 +14,8 @@ pub(crate) enum StringLiteralParentKind {
     Expression,
     /// Variant to track tokens that are inside a member
     Member,
+    /// Variant to track tokens that are inside an assert
+    Assert,
```
Comment (Conaclos) [inline crates/biome_js_formatter/src/utils/string_utils.rs:18]: Import assertion is the old name for import attributes. I think we should choose another variant name. Maybe:
```suggestion
    /// Variant to track tokens that are inside an import attribute
    ImportAttribute,
```

File: crates/biome_js_formatter/src/utils/string_utils.rs [hunk 2]
```diff
@@ -229,6 +232,17 @@ impl<'token> LiteralStringNormaliser<'token> {
         self.token.token()
     }
 
+    fn normalise_assert(&mut self, string_information: StringInformation) -> Cow<'token, str> {
```
Comment (Conaclos) [inline crates/biome_js_formatter/src/utils/string_utils.rs:235]:
```suggestion
    fn normalise_import_attribute(&mut self, string_information: StringInformation) -> Cow<'token, str> {
```

File: crates/biome_js_formatter/src/utils/string_utils.rs [hunk 3]
```diff
@@ -229,6 +232,17 @@ impl<'token> LiteralStringNormaliser<'token> {
...
+        if self.can_remove_assert_quotes() {
+            return Cow::Owned(self.raw_content().to_string());
+        }
+
+        self.normalise_string_literal(string_information)
+    }
+    fn can_remove_assert_quotes(&self) -> bool {
```
Comment (Conaclos) [inline crates/biome_js_formatter/src/utils/string_utils.rs:242]:
```suggestion
    fn can_remove_import_attribute_quotes(&self) -> bool {
```

**Review by Conaclos — COMMENTED** [review]
Looks good to me. I've just suggested a change of name because `import assertion` is the old name of `import attribute`.

**Comment by suxin2017** [comment]
> Looks good to me. I've just suggested a change of name because `import assertion` is the old name of `import attribute`.

Thank you for your review, I've done.

**Comment by Conaclos** [comment]
Everything is green, except rust code formatting. Once formatted, we can merge the PR.

**Pull request merged**